From 506f093cbd075df4491001d4b85f59e0d116f86e Mon Sep 17 00:00:00 2001
From: Osamu Aoki <osamu@debian.org>
Date: Fri, 29 Sep 2017 22:56:53 +0900
Subject: [PATCH] Bump -dev package name

Since this is not just EUC -> UTF-8 change but is a big change which requires
other dependency packages to be updated to get them build, let's bump *-dev
package to prevent FTBFS.  This change should allow anthy to move to testing
without waiting for all ibus/fcitx/uim updates to the new libraries.  Here,
coexistence with both version is the key.

Also this bumps soname of libanthyinput to avoid upgrade problem.

In this patch unused debian patches are removed.

Signed-off-by: Osamu Aoki <osamu@debian.org>
---
 debian/changelog                                   | 10 ++++++
 debian/control                                     | 14 +++-----
 debian/{libanthy-dev.docs => libanthy1-dev.docs}   |  0
 ...{libanthy-dev.install => libanthy1-dev.install} |  0
 ...nput-dev.install => libanthyinput1-dev.install} |  0
 ...banthyinput0.install => libanthyinput1.install} |  0
 .../patches/0000_libanthyinput_bump_soname.patch   | 18 ++++++++++
 debian/patches/10_Makefile.am_remove-elc           | 39 ----------------------
 debian/patches/series                              |  2 +-
 debian/patches/utf8_heartmark                      | 10 ------
 src-util/Makefile.am                               |  1 +
 11 files changed, 35 insertions(+), 59 deletions(-)
 rename debian/{libanthy-dev.docs => libanthy1-dev.docs} (100%)
 rename debian/{libanthy-dev.install => libanthy1-dev.install} (100%)
 rename debian/{libanthyinput-dev.install => libanthyinput1-dev.install} (100%)
 rename debian/{libanthyinput0.install => libanthyinput1.install} (100%)
 create mode 100644 debian/patches/0000_libanthyinput_bump_soname.patch
 delete mode 100644 debian/patches/10_Makefile.am_remove-elc
 delete mode 100644 debian/patches/utf8_heartmark

diff --git a/debian/changelog b/debian/changelog
index 76db720..bf476a3 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,13 @@
+anthy (1:0.3-5.1) unstable; urgency=medium
+
+  * Non-maintainer upload. 
+  * Fix upgrade problem by bumping soname.  Closes: #789927, #874708
+  * Made lib* and lib*-dev package name include soname major version
+    to enable smooth and slow migration without many FTBFS and install
+    problems.  Closes: #789927, #788253, #789914, #789927, #810351, #873860
+
+ -- Osamu Aoki <osamu@debian.org>  Sat, 30 Sep 2017 01:32:00 +0900
+
 anthy (1:0.3-5) unstable; urgency=medium
 
   * debian/control (Breaks, Replaces): Fix version specification using <<,
diff --git a/debian/control b/debian/control
index fd1e137..dea62fb 100644
--- a/debian/control
+++ b/debian/control
@@ -28,7 +28,7 @@ Description: Japanese kana-kanji conversion - dictionary
  hiragana text to mixed kana and kanji.  It is implemented as a library
  and stores private information in ~/.anthy/.
  .
- This package provides common dictonary data files.
+ This package provides common dictionary data files.
 
 Package: anthy-el
 Section: lisp
@@ -55,7 +55,7 @@ Description: Japanese kana-kanji conversion - runtime library
  .
  This package provides runtime shared libraries for Anthy.
 
-Package: libanthy-dev
+Package: libanthy1-dev
 Section: libdevel
 Architecture: any
 Multi-Arch: same
@@ -68,13 +68,11 @@ Description: Japanese kana-kanji conversion - runtime library
  This package provides the files necessary for compiling programs that
  use Anthy.
 
-Package: libanthyinput0
+Package: libanthyinput1
 Section: libs
 Architecture: any
 Multi-Arch: same
 Pre-Depends: ${misc:Pre-Depends}
-Replaces: libanthy0 (<< 1:0.3-1)
-Breaks: libanthy0 (<< 1:0.3-1)
 Depends: ${shlibs:Depends}, ${misc:Depends}
 Description: Japanese kana-kanji conversion - runtime library for input
  Anthy is kana-kanji conversion engine for Japanese.  It converts
@@ -83,13 +81,11 @@ Description: Japanese kana-kanji conversion - runtime library for input
  .
  This package provides shared library for input method.
 
-Package: libanthyinput-dev
+Package: libanthyinput1-dev
 Section: libdevel
 Architecture: any
 Multi-Arch: same
-Replaces: libanthy-dev (<< 1:0.3-1)
-Breaks: libanthy-dev (<< 1:0.3-1)
-Depends: libanthyinput0 (= ${binary:Version}), ${misc:Depends}
+Depends: libanthyinput1 (= ${binary:Version}), ${misc:Depends}
 Description: Japanese kana-kanji conversion - runtime library for input
  Anthy is kana-kanji conversion engine for Japanese.  It converts
  hiragana text to mixed kana and kanji.  It is implemented as a library
diff --git a/debian/libanthy-dev.docs b/debian/libanthy1-dev.docs
similarity index 100%
rename from debian/libanthy-dev.docs
rename to debian/libanthy1-dev.docs
diff --git a/debian/libanthy-dev.install b/debian/libanthy1-dev.install
similarity index 100%
rename from debian/libanthy-dev.install
rename to debian/libanthy1-dev.install
diff --git a/debian/libanthyinput-dev.install b/debian/libanthyinput1-dev.install
similarity index 100%
rename from debian/libanthyinput-dev.install
rename to debian/libanthyinput1-dev.install
diff --git a/debian/libanthyinput0.install b/debian/libanthyinput1.install
similarity index 100%
rename from debian/libanthyinput0.install
rename to debian/libanthyinput1.install
diff --git a/debian/patches/0000_libanthyinput_bump_soname.patch b/debian/patches/0000_libanthyinput_bump_soname.patch
new file mode 100644
index 0000000..11e8601
--- /dev/null
+++ b/debian/patches/0000_libanthyinput_bump_soname.patch
@@ -0,0 +1,18 @@
+Description: Bump soname also for libanthyinput
+ libanthyinput0 fails to upgrade from 'sid' since it trying 
+ to overwrite /usr/lib/x86_64-linux-gnu/libanthyinput.so.0.0.0.
+ Bumping soname as other libraries with this patch, this bug
+ is avoided.  Closes: #789927
+Author: osamu@debian.org
+Bug-Debian: https://bugs.debian.org/789927
+Last-Update: 2017-09-29
+
+--- anthy-0.3.orig/src-util/Makefile.am
++++ anthy-0.3/src-util/Makefile.am
+@@ -26,5 +26,6 @@ lib_LTLIBRARIES = libanthyinput.la
+ libanthyinput_la_SOURCES = input.c rkconv.c rkhelper.c\
+  rkconv.h rkmap.h rkhelper.h
+ libanthyinput_la_LIBADD = ../src-main/libanthy.la
++libanthyinput_la_LDFLAGS = -version-info 2:2:1
+ 
+ pkgdata_DATA = typetab dic-tool-usage.txt
diff --git a/debian/patches/10_Makefile.am_remove-elc b/debian/patches/10_Makefile.am_remove-elc
deleted file mode 100644
index b8277b1..0000000
--- a/debian/patches/10_Makefile.am_remove-elc
+++ /dev/null
@@ -1,39 +0,0 @@
-Index: anthy-9100h/src-util/Makefile.am
-===================================================================
---- anthy-9100h.orig/src-util/Makefile.am	2007-06-20 16:40:09.000000000 +0900
-+++ anthy-9100h/src-util/Makefile.am	2010-03-29 10:36:09.836718765 +0900
-@@ -8,7 +8,6 @@
-  anthy-isearch.el anthy-azik.el anthy-kyuri.el
- EXTRA_DIST = $(ELISP_FILES) typetab dic-tool-usage.txt anthy.i
- bin_PROGRAMS = anthy-dic-tool anthy-agent anthy-morphological-analyzer
--ELCFILES = anthy.elc anthy-dic.elc anthy-azik.elc anthy-conf.elc anthy-isearch.elc anthy-kyuri.elc leim-list.elc
- if ELISP
- lisp_LISP = $(ELISP_FILES)
- endif
-===================================================================
---- anthy-9100h.orig/src-util/Makefile.in	2009-02-07 21:21:39.000000000 +0900
-+++ anthy-9100h/src-util/Makefile.in	2010-05-14 18:41:18.000000000 +0900
-@@ -109,6 +109,7 @@
- am__ELFILES = anthy-azik.el anthy-conf.el anthy-dic.el \
- 	anthy-isearch.el anthy-kyuri.el anthy.el leim-list.el
- am__ELCFILES = $(am__ELFILES:.el=.elc)
-+ELCFILES = $(LISP:.el=.elc)
- elisp_comp = $(top_srcdir)/elisp-comp
- pkgdataDATA_INSTALL = $(INSTALL_DATA)
- DATA = $(pkgdata_DATA)
-@@ -167,6 +168,7 @@
- PACKAGE_NAME = @PACKAGE_NAME@
- PACKAGE_STRING = @PACKAGE_STRING@
- PACKAGE_TARNAME = @PACKAGE_TARNAME@
-+PACKAGE_URL = @PACKAGE_URL@
- PACKAGE_VERSION = @PACKAGE_VERSION@
- PATH_SEPARATOR = @PATH_SEPARATOR@
- RANLIB = @RANLIB@
-@@ -230,7 +232,6 @@
-  anthy-isearch.el anthy-azik.el anthy-kyuri.el
- 
- EXTRA_DIST = $(ELISP_FILES) typetab dic-tool-usage.txt anthy.i
--ELCFILES = anthy.elc anthy-dic.elc anthy-azik.elc anthy-conf.elc anthy-isearch.elc anthy-kyuri.elc leim-list.elc
- @ELISP_TRUE@lisp_LISP = $(ELISP_FILES)
- noinst_LTLIBRARIES = libconvdb.la
- libconvdb_la_SOURCES = convdb.c
diff --git a/debian/patches/series b/debian/patches/series
index 1bb8bf6..610714d 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -1 +1 @@
-# empty
+0000_libanthyinput_bump_soname.patch
diff --git a/debian/patches/utf8_heartmark b/debian/patches/utf8_heartmark
deleted file mode 100644
index 4b617f5..0000000
--- a/debian/patches/utf8_heartmark
+++ /dev/null
@@ -1,10 +0,0 @@
-Index: anthy-9100h/mkworddic/utf8.t
-===================================================================
---- anthy-9100h.orig/mkworddic/utf8.t	2010-05-01 20:25:42.000000000 +0900
-+++ anthy-9100h/mkworddic/utf8.t	2010-05-01 20:53:53.000000000 +0900
-@@ -2,4 +2,4 @@
- # example of utf8 dictionary.
- # please have a GETA ;-)
- #
--はーとまーく #T35 ❤
-+はーとまーく #T35 ♡ ♥
diff --git a/src-util/Makefile.am b/src-util/Makefile.am
index ca1ff71..93ead4f 100644
--- a/src-util/Makefile.am
+++ b/src-util/Makefile.am
@@ -26,5 +26,6 @@ lib_LTLIBRARIES = libanthyinput.la
 libanthyinput_la_SOURCES = input.c rkconv.c rkhelper.c\
  rkconv.h rkmap.h rkhelper.h
 libanthyinput_la_LIBADD = ../src-main/libanthy.la
+libanthyinput_la_LDFLAGS = -version-info 2:2:1
 
 pkgdata_DATA = typetab dic-tool-usage.txt
-- 
2.14.2

